src/main/java/com/metabug/web/controller/IndexController.java

```diff
@@ -4,13 +4,19 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.servlet.ModelAndView;
 
+import java.security.Principal;
+
 @Controller
 public class IndexController {
 
-    @GetMapping(value = {"/login"})
-    public ModelAndView index() {
+    @GetMapping(value = {"/login", ""})
```

Comment:
Why `""`?

Reply:
Login page is on both /login and "", so if "" is not present, a logged user could access the login page.

Reply:
Login page is only on `/login`. Spring Security automatically creates filter chain to    redirect `/login`, if you want `""` to be accessible as login page, in my understanding you have to add `.loginPage("")` in configuration 

WebSecurityConfiguration:
```java
...
                .formLogin()
           ---> .loginPage("/login")            
                .failureUrl("/login?error=true")
                .defaultSuccessUrl("/home")
                .usernameParameter("login")
                .passwordParameter("password")
...
```

Reply:
Then explain this.
![image](https://user-images.githubusercontent.com/58629803/80597522-4211a900-8a28-11ea-83da-16e1566ab2d7.png)

Reply:
Got it, resolved.